### PR TITLE
Lumberjack tweaks

### PR DESCRIPTION
**Pull request by white238**
* Fixes memory leak in lumberjack's MPI communication
* Clarifies message count is not just rank count
* Increases information out of LumberjackStream in regards to rank and message count

### CODE REVIEW

File: src/axom/slic/streams/LumberjackStream.hpp [hunk 1]
```diff
@@ -41,17 +41,9 @@ namespace slic
 /*!
  * \class LumberjackStream
  *
- * \brief A concrete instance of LogStream that dumps messages to a C++
- *  std::ostream object.
+ * \brief A concrete instance of LogStream that utilizes Lumberjack to
+ *  filter and pass messagse between MPI nodes.
```
Comment (kennyweiss) [inline src/axom/slic/streams/LumberjackStream.hpp]: Typo: messagse -> messages

File: src/axom/lumberjack/MPIUtility.cpp [hunk 2]
```diff
@@ -49,7 +51,8 @@ void mpiNonBlockingSendMessages(MPI_Comm comm, int destinationRank,
   MPI_Request mpiRequest;
   MPI_Isend(const_cast<char*>(packedMessagesToBeSent),
             strlen(packedMessagesToBeSent), MPI_CHAR,
-            destinationRank, 0, comm, &mpiRequest);
+            destinationRank, LJ_TAG, comm, &mpiRequest);
+  MPI_Request_free(&mpiRequest);
```
Comment (rhornung67) [inline src/axom/lumberjack/MPIUtility.cpp]: Good catch!

File: src/axom/lumberjack/MPIUtility.cpp [hunk 3]
```diff
@@ -22,23 +22,25 @@ namespace axom
...
   charArray = new char[messageSize+1];
   charArray[messageSize] = '\0';
 
   // Receive packed Message
-  MPI_Recv(charArray, messageSize, MPI_CHAR, mpiStatus.MPI_SOURCE, 0, comm,
-           &mpiStatus);
+  MPI_Recv(charArray, messageSize, MPI_CHAR, mpiStatus.MPI_SOURCE,
+           LJ_TAG, comm, &mpiStatus);
 
   return charArray;
```
Comment (gzagaris) [inline src/axom/lumberjack/MPIUtility.cpp]: How is `charArray` deleted?

File: src/axom/slic/streams/LumberjackStream.hpp (hunk 1 above)
Comment (white238) [reply src/axom/slic/streams/LumberjackStream.hpp]: done

File: src/axom/lumberjack/MPIUtility.cpp (hunk 2 above)
Comment (white238) [reply src/axom/lumberjack/MPIUtility.cpp]: @kennyweiss finally figured it out in a long won battle to get me out of his office...

File: src/axom/lumberjack/MPIUtility.cpp (hunk 3 above)
Comment (white238) [reply src/axom/lumberjack/MPIUtility.cpp]: It's deleted by the Lumberjack::pushMessagesOnce() or Lumberjack::pushMessagesFully()

File: src/axom/lumberjack/MPIUtility.cpp [hunk 4]
```diff
@@ -22,23 +22,25 @@ namespace axom
 namespace lumberjack
 {
 
+const int LJ_TAG = 55432;
```
[the remark this answers is not in the document]
Comment (white238) [reply src/axom/lumberjack/MPIUtility.cpp]: Done.

File: src/axom/lumberjack/MPIUtility.cpp (hunk 2 above)
Comment (cyrush) [reply src/axom/lumberjack/MPIUtility.cpp]: awesome you tracked down the issue!

**Pull request merged**